model/buildlogger.go

```diff
@@ -425,15 +426,34 @@ func (l *Logs) Find(ctx context.Context, opts LogFindOptions) error {
 	}
 
 	l.populated = false
-	findOpts := options.Find()
-	if opts.Limit > 0 {
-		findOpts.SetLimit(opts.Limit)
-	}
-	findOpts.SetSort(bson.D{
+
+	var it *mongo.Cursor
+	var err error
+	query := createFindQuery(opts)
+	sort := bson.D{
 		{Key: logInfoExecutionKey, Value: -1},
 		{Key: logCreatedAtKey, Value: -1},
-	})
-	it, err := l.env.GetDB().Collection(buildloggerCollection).Find(ctx, createFindQuery(opts), findOpts)
+	}
+	if opts.Group != "" && len(opts.Info.Tags) > 0 {
+		pipeline := []bson.M{
+			bson.M{"$match": query},
```

Comment:
nit: you don't need to declare all the `bson.M` types when initializing the slice elements since the compiler already knows they're of type `bson.M` (e.g. this could just be `{"$match": query}`).